File: packages/ts-sdk/src/treedb/adapters/graph.ts
import { jsonRequest, segment, type TreeDbAdapterContext } from './common.js';

export class GraphAdapter {
  constructor(private readonly context: TreeDbAdapterContext) {}
  refresh(repoId: string, input?: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/graph/refresh`, input); }
  query(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/graph/query`, input); }
  refreshJob(repoId: string, jobId: string): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}/graph/refresh-jobs/${segment(jobId)}`); }
  node(repoId: string, nodeId: string, query?: Record<string, string | number | boolean | undefined>): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}/graph/nodes/${segment(nodeId)}`, undefined, query); }
  related(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/graph/related`, input); }
  subgraph(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/graph/subgraph`, input); }
  searchFiles(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/graph/search-files`, input); }
  searchSections(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/graph/search-sections`, input); }
  searchEntities(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/graph/search-entities`, input); }
}

File: packages/rust-sdk/src/adapters/graph.rs
use std::sync::Arc;

use serde_json::Value;

use crate::adapters::common::{json_request, segment};
use crate::error::TreeDxResult;
use crate::transport::{Transport, TreeDxHttpMethod};

#[derive(Clone)]
pub struct GraphAdapter {
    transport: Arc<dyn Transport>,
}

impl GraphAdapter {
    pub fn new(transport: Arc<dyn Transport>) -> Self {
        Self { transport }
    }

    pub async fn refresh(&self, repo_id: &str, body: Value) -> TreeDxResult<Value> {
        json_request(
            &self.transport,
            TreeDxHttpMethod::Post,
            format!("/api/v1/repos/{}/graph/refresh", segment(repo_id)),
            Some(body),
            None,
        )
        .await
    }

    pub async fn query(&self, repo_id: &str, body: Value) -> TreeDxResult<Value> {
        json_request(
            &self.transport,
            TreeDxHttpMethod::Post,
            format!("/api/v1/repos/{}/graph/query", segment(repo_id)),
            Some(body),
            None,
        )
        .await
    }
}
